<launch>
    <arg name="ugv_num" default="3" />
    <arg name="ugv_id" default="1" />
    <arg name="ugv_name" default="ugv" />
    <arg name="flag_printf" default="true" />
    <arg name="enable_rviz" default="true" />
    <arg name="simulation_mode" default="true" />
    <arg name="location_source" default="0" />
    <arg name="odom_topic" default="" />
    <arg name="vel_topic" default="" />

    <arg name="enable_astar" default="false" />
    <arg name="laser_topic" default="/scan" />
    <arg name="octomap_topic" default="/octomap_full" />
    <arg name="occupancy_topic" default="/map" />
    <arg name="map/resolution" default="0.4" />
    <arg name="map/inflate" default="0.3" />

    <!-- 控制参数 -->
    <arg name="ugv_control_param/Kp_xy" default="2.0" />
    <arg name="ugv_control_param/Kp_yaw" default="0.95" />
    <arg name="ugv_control_param/max_vel_xy" default="1.5" />
    <arg name="ugv_control_param/max_vel_yaw" default="1.57" />
    <!-- 地理围栏参数 -->
    <arg name="ugv_geo_fence/max_x" default="12.0" />
    <arg name="ugv_geo_fence/min_x" default="-12.0" />
    <arg name="ugv_geo_fence/max_y" default="12.0" />
    <arg name="ugv_geo_fence/min_y" default="-12.0" />

    <node name="ugv_control_node_$(arg ugv_id)" pkg="sunray_ugv_control" type="ugv_control_node" output="screen">
        <param name="ugv_id" value="$(arg ugv_id)" />
        <param name="ugv_name" value="$(arg ugv_name)" />
        <param name="flag_printf" value="$(arg flag_printf)" />
        <param name="enable_rviz" value="$(arg enable_rviz)" />
        <!-- 定位 驱动话题 -->
        <param name="location_source" value="$(arg location_source)" />
        <param name="odom_topic" value="$(arg odom_topic)" />
        <param name="vel_topic" value="$(arg vel_topic)" />
        <!-- 控制参数 -->
        <param name="ugv_control_param/Kp_xy" value="$(arg ugv_control_param/Kp_xy)" />
        <param name="ugv_control_param/Kp_yaw" value="$(arg ugv_control_param/Kp_yaw)" />
        <param name="ugv_control_param/max_vel_xy" value="$(arg ugv_control_param/max_vel_xy)" />
        <param name="ugv_control_param/max_vel_yaw" value="$(arg ugv_control_param/max_vel_yaw)" />
        <!-- 地理围栏参数 -->
        <param name="ugv_geo_fence/max_x" value="$(arg ugv_geo_fence/max_x)" />
        <param name="ugv_geo_fence/min_x" value="$(arg ugv_geo_fence/min_x)" />
        <param name="ugv_geo_fence/max_y" value="$(arg ugv_geo_fence/max_y)" />
        <param name="ugv_geo_fence/min_y" value="$(arg ugv_geo_fence/min_y)" />
        <!-- 地图参数 -->
        <param name="enable_astar" value="$(arg enable_astar)" />
        <param name="map/resolution" value="$(arg map/resolution)" />
        <param name="map/inflate" value="$(arg map/inflate)" />
        <!-- 仿真模式参数 -->
        <param name="simulation_mode" value="$(arg simulation_mode)" />
        <!-- 雷达话题参数 -->
        <param name="laser_topic" value="$(arg laser_topic)" />
        <!-- 地图相关话题参数 -->
        <param name="octomap_topic" value="$(arg octomap_topic)" />
        <param name="occupancy_topic" value="$(arg occupancy_topic)" />
    </node>

    <include file="$(find sunray_ugv_control)/launch/ugv_control_swarm.launch" if="$(eval (arg('ugv_num') > arg('ugv_id')))">
        <arg name="ugv_id" value="$(eval arg('ugv_id') + 1)" />
        <arg name="ugv_num" value="$(arg ugv_num)" />
    </include>
</launch>
